### PR TITLE
Remove aes_x86core.c as it is unused

### LINKED ISSUE #20036: OpenSSL 3.0: Compiler error with gcc 12.2.0 for x86_64 and lp64 data model
With OpenSSL 3.0, I get a compile error in **aes_x86core.c**, when using gcc 12.2.0 for a x86_64 target and lp64 data model.
It worked with OpenSSL 1.1.1n

```
crypto/aes/aes_x86core.c:89:28: error: conflicting types for 'u64'; have 'long long unsigned int'

crypto/aes/aes_local.h:27:18: note: previous declaration of 'u64' with type 'u64' {aka 'long unsigned int'}
```

there are two different typedefs for u64:

aes_local.h:
`typedef uint64_t u64;` 

aes_x86core.h:
```
#if (defined(_WIN32) || defined(_WIN64)) && !defined(__MINGW32__)
typedef unsigned __int64 u64;
#define U64(C)  C##UI64
#elif defined(__arch64__)
typedef unsigned long u64;
#define U64(C)  C##UL
#else
typedef unsigned long long u64;
#define U64(C)  C##ULL
#endif
```

for x86_64 with lp64 data model, aes_x86core.h defines **unsigned long long** for u64, which is wrong and conflicts with the type defined in aes_local.h

Is this a bug or is there something wrong with my configuration?

### PR DESCRIPTION
**Pull request by t8m**
Fixes #20036

### CODE REVIEW

**Review by paulidale — APPROVED** [review]

File: crypto/aes/aes_x86core.c [hunk 1]
```diff
@@ -80,17 +80,6 @@ static void prefetch256(const void *table)
 #undef GETU32
 #define GETU32(p) (*((u32*)(p)))
 
-#if (defined(_WIN32) || defined(_WIN64)) && !defined(__MINGW32__)
```
Comment (slontis) [inline crypto/aes/aes_x86core.c:83]: I think I added this for originally to fix a problem with AIX truncation..
These defines look like they were copied from modes.h?
Where does the U64() come from with this change?

**Comment by t8m** [comment]
Not sure if this should be applied to 3.1 and 3.0 branches as this is just a cleanup.

**Review by paulidale — APPROVED** [review]

**Comment by bernd-edlinger** [comment]
Note I seem to remember, that this is the "reference" C implementation for x86[_64] assembler modules.

**Comment by slontis** [comment]
Not sure that it should be removed.

**Comment by t8m** [comment]
Keeping unused and unmaintained source in the tree is simply wrong.

**Review by hlandau — APPROVED** [review]

**Comment by t-j-h** [comment]
I think the issue here is that it should be kept in the tree *and* there should be a CI target that actually uses it.
Uncompiled code isn't all that useful - but this does serve a purpose - and that we can go back in git and find it really isn't the solution for this. We don't want this code to "rot" as such so it needs to be compiled and tested as otherwise it should be removed.

**Comment by t8m** [comment]
> I think the issue here is that it should be kept in the tree _and_ there should be a CI target that actually uses it.

It's not just that there is no CI target that would build it. The file is completely unused by the build.info files. I do not think it would be a good idea to create some completely arbitrary an non-default build targets that would use this source file.

**Comment by levitte** [comment]
> > I think the issue here is that it should be kept in the tree _and_ there should be a CI target that actually uses it.
> 
> It's not just that there is no CI target that would build it. The file is completely unused by the build.info files. I do not think it would be a good idea to create some completely arbitrary and non-default build targets that would use this source file.

It's been used to generate assembly files for new assemblers (assembler compilers?  I forget the proper terminology), to serve as a template for perlasm xlate code.  That's at least what I recall.

It *should* be possible to compile directly on appropriate hardware, but in terms of OpenSSL configuration, we don't have a clear way to get it selected, as it would seem wrong to use it with `no-asm`, but also without `no-asm` since that's where we use the perlasm stuff that's once been based on this file...
so to do this sanely, we'd need another (dev only?) configuration option...  `no-asm-or-almost`?

**Comment by t8m** [comment]
**OTC: We want to make the code built and maintained.**

**Pull request closed without merging**